CMakeLists.txt

```diff
@@ -201,9 +201,9 @@ elseif(GNU_GCC OR LLVM_CLANG)
         # TODO(rryan): macOS can use SSE3, and possibly SSE 4.1 once
         # we require macOS 10.12.
         # https://stackoverflow.com/questions/45917280/mac-osx-minumum-support-sse-version
-      elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "arm")
+      elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^arm.*")
```

Comment:
This regex does not work now, because it matches a "armv8b" which is a arm64. This means it matches some arm64. 
Please revert to the original and add a comment why this is correct. 
Something like: 
// CMAKE_SYSTEM_PROCESSOR is just "arm" for all 32 bit ARMs. These flags are not valid with AArch64 and ARMv8 targets where CMAKE_SYSTEM_PROCESSOR is one of ("aarch64_be (arm64)", "aarch64 (arm64)", "armv8b (arm64 compat)" and "armv8l (arm64 compat)".